packages/array-upsert/index.d.ts

```diff
@@ -0,0 +1,3 @@
+declare function upsert<T>(arr: T[], newValue: T, targetIndex: number): T[];
```

Comment:
I don't think we should limit new value to type of existing elements.
So `upsert([1,2,3], 'a', 2)` should be valid